src/function/scalar/list/list_reverse.cpp

```diff
@@ -0,0 +1,124 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_parameter_expression.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+
+namespace duckdb {
+
+static void ListReverseFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	auto count = args.size();
+
+	Vector &input_list = args.data[0];
+	if (input_list.GetType().id() == LogicalTypeId::SQLNULL) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		ConstantVector::SetNull(result, true);
+		return;
+	}
+
+	// get the child vector
+	auto input_list_list_size = ListVector::GetListSize(input_list);
+	auto &input_list_child = ListVector::GetEntry(input_list);
+
+	UnifiedVectorFormat input_list_data;
+	input_list.ToUnifiedFormat(count, input_list_data);
+	auto input_list_entries = (list_entry_t *)input_list_data.data;
+
+	// get the child data
+	UnifiedVectorFormat input_list_child_data;
+	input_list_child.ToUnifiedFormat(input_list_list_size, input_list_child_data);
+
+	result.SetVectorType(VectorType::FLAT_VECTOR);
+	auto result_entries = FlatVector::GetData<list_entry_t>(result);
+	auto &result_validity = FlatVector::Validity(result);
+
+	// create a selection vector for slicing the child vector
+	SelectionVector rev_sel(input_list_list_size);
+
+	idx_t offset = 0;
+	idx_t offset_per_new_row = 0;
+
+	for (idx_t i = 0; i < count; i++) {
+		auto input_list_list_index = input_list_data.sel->get_index(i);
+
+		if (!input_list_data.validity.RowIsValid(input_list_list_index)) {
+			result_validity.SetInvalid(i);
```

Comment:
because we don't set the result validity manually, this is also not needed

Reply:
It appears that we need to do that to handle cases like ```select list_reverse(columns(['int_array', 'varchar_array', 'nested_int_array', 'array_of_structs', 'timestamp_array', 'double_array', 'date_array', 'timestamptz_array'])) from test_all_types();
```